master/internal/user/postgres_users_intg_test.go

```diff
@@ -0,0 +1,358 @@
+		log.Panicln(err)
+	}
+
+	os.Exit(m.Run())
+}
+
+// Tests for postgres_users.go.
+func TestUserAdd(t *testing.T) {
+	var (
+		resAUG      model.AgentUserGroup
+		resGroup    model.Group
+		resGroupMem model.GroupMembership
+	)
+	cases := []struct {
+		name string
+		user model.User
+		ug   model.AgentUserGroup
+	}{
+		{
+			"simple-case",
+			model.User{Username: uuid.NewString()},
+			model.AgentUserGroup{},
+		},
+		{
+			"aug-defined",
+			model.User{Username: uuid.NewString()},
+			model.AgentUserGroup{User: uuid.NewString(), UID: 1},
+		},
+	}
+	for _, tt := range cases {
+		t.Run(tt.name, func(t *testing.T) {
+			// Test Add.
+			uid, err := Add(context.TODO(), &tt.user, &tt.ug)
+			require.NoError(t, err)
+
+			// require entry in the users table (addUser).
+			var resUser model.User
+
+			err = db.Bun().NewSelect().Table("users").Where("id = ?", uid).Scan(context.TODO(), &resUser)
+			require.NoError(t, err)
+			// Reset the modified at variable, just for testing purposes.
+			resUser.ModifiedAt = tt.user.ModifiedAt
+			require.Equal(t, tt.user, resUser)
+
+			// require entry in the group table (addUser).
+			err = db.Bun().NewSelect().Table("groups").
+				Where("user_id = ?", uid).Scan(context.TODO(), &resGroup)
+			require.NoError(t, err)
+			require.Equal(t, uid, resGroup.OwnerID)
+			require.Equal(t, fmt.Sprintf("%d%s", uid, PersonalGroupPostfix), resGroup.Name)
+
+			// require entry in the group membership table (addUser).
+			err = db.Bun().NewSelect().Table("user_group_membership").
+				Where("user_id = ?", uid).Scan(context.TODO(), &resGroupMem)
+			require.NoError(t, err)
+			require.Equal(t, uid, resGroupMem.UserID)
+			require.Equal(t, resGroup.ID, resGroupMem.GroupID)
+
+			// require entry in the AUG membership table (addAgentUserGroup).
+			// But also check that the AUG isn't nil/empty.
+			if tt.ug != (model.AgentUserGroup{}) {
+				err = db.Bun().NewSelect().Table("agent_user_groups").
+					Where("user_id = ?", uid).Scan(context.TODO(), &resAUG)
+				require.NoError(t, err)
+				require.Equal(t, uid, resAUG.UserID)
+				require.Equal(t, tt.ug.User, resAUG.User)
+				require.Equal(t, tt.ug.UID, resAUG.UID)
+			}
+		})
+	}
+}
+
+func TestUserUpdate(t *testing.T) {
+	cases := []struct {
+		name        string
+		ug          model.AgentUserGroup
+		updatedUser model.User
+	}{
+		{
+			"simple-case",
+			model.AgentUserGroup{},
+			model.User{Username: uuid.NewString()},
+		},
+		{
+			"aug-defined-1",
+			model.AgentUserGroup{User: uuid.NewString(), UID: 1},
+			model.User{Username: uuid.NewString(), Admin: false, Active: false, Remote: false},
+		},
+		{
+			"aug-defined-2",
+			model.AgentUserGroup{User: uuid.NewString(), UID: 123},
+			model.User{Username: uuid.NewString(), Admin: true, Active: true, Remote: true},
+		},
+	}
+	for _, tt := range cases {
+		t.Run(tt.name, func(t *testing.T) {
+			user, err := addTestUser(nil)
+			require.NoError(t, err)
+
+			// Test Update.
+			tt.updatedUser.ID = user.ID
```

Comment:
maybe could set ```tt.updatedUser.DisplayName = "dontupdate"```, and assert that it isn't updated because it isn't in the field mask 

displayName is just an example for a field